uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-legend.component.js

```diff
@@ -33,19 +37,72 @@ define([
       { label: 'AT', description: 'Accrued TOIL' }
     ];
 
+    vm.checkIfAbsenceTypeIsSelectedForFiltering = checkIfAbsenceTypeIsSelectedForFiltering;
     vm.getAbsenceTypeStyle = getAbsenceTypeStyle;
+    vm.resetFilteringByAbsenceTypes = resetFilteringByAbsenceTypes;
+    vm.toggleFilteringByAbsenceType = toggleFilteringByAbsenceType;
+
+    (function init () {
+      initWatchers();
+    }());
+
+    /**
+     * Checks if absence type is selected for filtering
+     *
+     * @param  {String} absenceTypeId
+     * @return {Boolean}
+     */
+    function checkIfAbsenceTypeIsSelectedForFiltering (absenceTypeId) {
+      return !vm.absenceTypesToFilterBy.length ||
+        _.includes(vm.absenceTypesToFilterBy, absenceTypeId);
```

Comment:
I think naming the conditions makes it more readable and helps explain why the condition is there:

```js
var isIncludedInTheAbsenceTypeFilters = _.includes(vm.absenceTypesToFilterBy, absenceTypeId);
var noAbsenceTypesFiltersHaveBeenSelected = !vm.absenceTypesToFilterBy.length;

return noAbsenceTypesFiltersHaveBeenSelected || isIncludedInTheAbsenceTypeFilters;
```

For example, it explains that this absence type is included in the filters by default if no specific absence type has been selected.

Reply:
Done